### PR TITLE
Replace the FakeWarden with the real one

### PR DESCRIPTION
**Pull request by acaron**
I used the spec helpers from Warden since it doesn't offer it has a test helper... It must be the real Warden if we want to add other options such as scope or strategies and test them!

### CODE REVIEW

File: test/test_helper.rb [hunk 1]
```diff
@@ -2,3 +2,79 @@
 require 'tyrant'
 
 require 'minitest/autorun'
+
+require 'warden'
+require 'rack-minitest/test'
+
+# Use Warden spec helper to simulate Warden
+# https://github.com/hassox/warden/blob/master/spec/helpers/request_helper.rb
+module Warden::Spec
```
Comment (apotonick) [inline test/test_helper.rb]: Don't you think we could use this module directly from Warden, without copying it?

Comment (acaron) [inline test/test_helper.rb]: It is in their Spec helpers so their own tests... I think we can only access /lib. If you know how, telle me :)

Comment (acaron) [inline test/test_helper.rb]: They have helpers for test, we should ask to add this kind of thing.

Comment (apotonick) [inline test/test_helper.rb]: Yeah, we have to ask them to make that helper available via `lib/test/helpers.rb`, we do the same in Cells! Do you want to PR them?

Comment (acaron) [inline test/test_helper.rb]: I'll write an issue to know how they want to manage this (probably don't want to copy the whole class) and propose my warden method too. We should like into Devise test to see how they manage that, but Devise code is frightening. So huge!

Comment (acaron) [inline test/test_helper.rb]: Look*

Comment (acaron) [inline test/test_helper.rb]: https://github.com/hassox/warden/issues/126

**Comment by acaron** [comment]
Pull request done: https://github.com/hassox/warden/pull/127

Once it is release, I'll change it!

**Comment by apotonick** [comment]
This is great Annie, thanks so much! So I'll simply wait on this one?

**Comment by acaron** [comment]
They are fast to answer so maybe wait is a good idea!

**Comment by acaron** [comment]
@apotonick it will be release with version 1.2.5 in Warden!

**Comment by acaron** [comment]
@apotonick Warden 1.2.6 is out. Check the final files changes!

**Comment by apotonick** [comment]
Thank you soooo much for looking after this, that  is a huge help! :heart:

**Pull request merged**